src/hooks/useContract.ts

```diff
@@ -45,16 +45,19 @@ export function useContract(
 ): Contract | null {
   const { account, isEVM } = useActiveWeb3React()
   const { library } = useWeb3React()
+  const { readProvider } = useKyberSwapConfig()
+
+  const lib = useMemo(() => (account ? library : readProvider), [account, library, readProvider])
```

Comment:
This seems not right. It will make every data calls go through wallet instead of Kyber's rpc?

Reply:
this is useContract hook. we have useContractForReading that use kyber rpc.
if wallet does not connected, library is null this contract is always return null -> this is totally wrong

Reply:
You re right, I read it mistakely